src/components/organisms/WorldEntranceForm/WorldEntranceForm.tsx

```diff
@@ -42,125 +41,88 @@ export const WorldEntranceForm: React.FC<WorldEntranceFormProps> = ({
-    clearDirty: clearDirtyCode,
-  } = useArray<Question>(world.questions?.code);
-
-  const {
-    items: profileQuestions,
-    add: addProfileQuestion,
-    update: updateProfileQuestion,
-    clear: clearProfileQuestions,
-    remove: removeProfileQuestion,
-    isDirty: isDirtyProfile,
-    clearDirty: clearDirtyProfile,
-  } = useArray<Question>(world.questions?.profile);
-
-  const {
-    items: entranceSteps,
-    add: addEntranceStep,
-    update: updateEntranceStep,
-    clear: clearEntranceSteps,
-    remove: removeEntranceStep,
-    isDirty: isDirtyEntrance,
-    clearDirty: clearDirtyEntrance,
-  } = useArray<EntranceStepConfig>(world.entrance, {
-    create: () => ({ template: EntranceStepTemplate.WelcomeVideo }),
-    prepare: (item) => ({
-      ...item,
-      template: item.template || EntranceStepTemplate.WelcomeVideo,
-    }),
-  });
-
   const defaultValues = useMemo<WorldEntranceFormInput>(
     () => ({
-      code: codeQuestions,
-      profile: profileQuestions,
-      entrance: entranceSteps,
+      code: world.questions?.code ?? [],
+      profile: world.questions?.profile ?? [],
+      entrance: world.entrance,
       adultContent: world.adultContent ?? false,
       requiresDateOfBirth: world.requiresDateOfBirth ?? false,
     }),
     [
-      codeQuestions,
-      profileQuestions,
-      entranceSteps,
+      world.questions?.code,
+      world.questions?.profile,
+      world.entrance,
       world.adultContent,
       world.requiresDateOfBirth,
     ]
   );
 
   const {
-    getValues,
     reset,
     register,
-    formState: { dirty, isSubmitting },
-    errors,
     handleSubmit,
+    control,
   } = useForm<WorldEntranceFormInput>({
     mode: "onSubmit",
     reValidateMode: "onChange",
-    validationSchema: worldEntranceSchema,
+    resolver: yupResolver(worldEntranceSchema),
     defaultValues,
   });
 
+  const {
+    fields: codeQuestions,
+    append: addCodeQuestion,
+    remove: removeCodeQuestion,
+  } = useFieldArray({ control, shouldUnregister: true, name: "code" });
+
+  const handleAddCodeQuestion = () =>
+    addCodeQuestion({ name: "", text: "", link: "" });
+  const clearCodeQuestions = () => removeCodeQuestion();
+
+  const {
+    fields: profileQuestions,
+    append: addProfileQuestion,
+    remove: removeProfileQuestion,
+  } = useFieldArray({ control, shouldUnregister: true, name: "profile" });
+
+  const handleAddProfileQuestion = () =>
+    addProfileQuestion({ name: "", text: "", link: "" });
+  const clearProfileQuestions = () => removeProfileQuestion();
+  const {
+    fields: entranceSteps,
+    append: addEntranceStep,
+    remove: removeEntranceStep,
+  } = useFieldArray({ control, shouldUnregister: true, name: "entrance" });
+
+  const handleAddEntranceQuestion = () =>
+    addEntranceStep({
+      template: EntranceStepTemplate.WelcomeVideo,
+      videoUrl: "",
+      autoplay: false,
+      welcomeText: "",
+    });
+  const clearEntranceSteps = () => removeEntranceStep();
```

Comment:
this thing has to accept no arguments to actually `clear` all fields, if used as `clearEntranceSteps removeEntranceStep` it will be accepting `event`, which might interfere with the result that we want to get

Reply:
just a handful of bad ideas:
- we put `as () => void` after it
- we put `useCallback` on it
- we leave it as it is for now

I had to mention it, but we don't have to waste time looking for a perfect solution.
Let's leave the code as it is.